sound/soc/sof/ipc4-pcm.c

```diff
@@ -21,12 +21,18 @@
  * @dai_copier: the dai copier of the pcm stream
  * @stream_start_offset: reported by fw in memory window
  * @llp_offset: llp offset in memory window
+ * @boundary: wrap boundary should be used for the LLP frame counter
+ * @delay: Calculated and stored in pointer callback to be returned as delay
+ *	   value
  */
 struct sof_ipc4_timestamp_info {
 	struct sof_ipc4_copier *host_copier;
 	struct sof_ipc4_copier *dai_copier;
 	u64 stream_start_offset;
 	u32 llp_offset;
+
+	u64 boundary;
+	snd_pcm_sframes_t delay;
```

Comment:
add comment along the lines of "negative values shall be ignored"

Reply:
The delay is going to be >= when it is calculated, otherwise it remains -EOPNOTSUPP to indicate that the delay is not calculated, which does not make much sense, does it?
I just set initialize it to 0 and return it in the delay callback....